qa/integration-tests/src/test/java/io/camunda/it/exporter/UserTaskExporterIT.java

```diff
@@ -0,0 +1,409 @@
+      // when
+      final var processInstanceId = startZeebeUserTaskProcess(client, null);
+
+      var userTasks = fetchUserTasks(client, Long.parseLong(processInstanceId));
+
+      client
+          .newUserTaskAssignCommand(userTasks.getFirst().getUserTaskKey())
+          .assignee("demo")
+          .send()
+          .join();
+
+      // then
+      waitForTask(
+          client,
+          f -> {
+            f.processInstanceKey(Long.parseLong(processInstanceId));
+            f.assignee("demo");
+          });
+
+      userTasks = fetchUserTasks(client, Long.parseLong(processInstanceId));
+      assertThat(userTasks).hasSize(1);
+      assertThat(userTasks.getFirst().getAssignee()).isEqualTo("demo");
+    }
+
+    @TestTemplate
+    void shouldUnassignUserTask(final TestStandaloneBroker testBroker) {
+      // given
+      final var client = testBroker.newClientBuilder().build();
+
+      // when
+      final var processInstanceId = startZeebeUserTaskProcess(client, t -> t.zeebeAssignee("demo"));
+
+      final var userTasks = fetchUserTasks(client, Long.parseLong(processInstanceId));
+
+      client.newUserTaskUnassignCommand(userTasks.getFirst().getUserTaskKey()).send().join();
+
+      // then
+      Awaitility.await()
+          .ignoreExceptions()
+          .timeout(Duration.ofSeconds(30))
+          .until(
+              () -> {
+                final var tasks =
+                    client
+                        .newUserTaskQuery()
+                        .filter(f -> f.processInstanceKey(Long.valueOf(processInstanceId)))
+                        .send()
+                        .join()
+                        .items();
+                return tasks.getFirst().getAssignee() == null;
+              });
+
+      final var unassignedTasks = fetchUserTasks(client, Long.parseLong(processInstanceId));
+      assertThat(unassignedTasks).hasSize(1);
+      assertThat(unassignedTasks.getFirst().getAssignee()).isNull();
+    }
+  }
+
+  @Nested
+  class UserTaskStateTransitionIT {
+
+    @TestTemplate
+    void shouldCompleteUserTask(final TestStandaloneBroker testBroker) {
+      // given
+      final var client = testBroker.newClientBuilder().build();
+
+      // when
+      final var processInstanceId = startZeebeUserTaskProcess(client, null);
+
+      var userTasks = fetchUserTasks(client, Long.parseLong(processInstanceId));
+
+      client.newUserTaskCompleteCommand(userTasks.getFirst().getUserTaskKey()).send().join();
+      // then
+      waitForTask(
+          client,
+          f -> {
+            f.processInstanceKey(Long.parseLong(processInstanceId));
+            f.state("COMPLETED");
+          });
+
+      userTasks = fetchUserTasks(client, Long.parseLong(processInstanceId));
+      assertThat(userTasks).hasSize(1);
+      assertThat(userTasks.getFirst().getState()).isEqualTo("COMPLETED");
+      assertThat(userTasks.getFirst().getCompletionDate()).isNotNull();
+    }
+  }
+
+  @Nested
+  class UserTaskWithFormsIT {
```

Comment:
Are you sure that this is the proper IT test for exporter? You also test `getExternalFormReference()` and `getFormKey()` in unit tests. Won't it be enough for exporter test?

Reply:
Mostly written then test for the FormKey scenario, since when creating a UserTask you specify the form with it's formId (human readable name) but on the index we only care for the formKey (UUID)